miqcli/api.py

```diff
@@ -28,125 +29,194 @@
 class ClientAPI(object):
     """ManageIQ API client class.
 
-    Class interfaces with ManageIQ server. The underlying component performing
-    the interactions with the server is the ManageIQ Python API Client.
+    Class interfaces with ManageIQ server. The underlying component
+    performing the interactions with the server is the ManageIQ Python
+    API Client.
 
     https://github.com/ManageIQ/manageiq-api-client-python
 
-    This library is used to establish connection to the server defined, talk
-    with all the servers collections and perform CRUD operations.
+    This library is used to establish connection to the server defined,
+    talk with all the servers collections and perform CRUD operations.
 
-    Each CLI command will reference this class to perform its collections
-    actions.
+    Each CLI command will reference this class to perform its
+    collections actions.
     """
 
     def __init__(self, settings):
         """Constructor."""
-        self._settings = settings
+
+        # create miqcli folder if it doesn't exist
+        try:
+            os.makedirs(os.path.dirname(AUTHDIR))
+        except OSError as e:
+            if e.errno != errno.EEXIST:
+                raise RuntimeError('Error creating user miqcli folder.')
+
+        # url details
+        self._url = settings.get('url', None)
+        if self._url is None:
+            self._url = 'https://localhost:8443'
+        self._url = self._url + '/api'
+        self._verify_ssl = settings.get('enable_ssl_verify', False)
```

Comment:
Also shouldn't enable_ssl_verify, username, and password default values come from DEFAULT_CONFIG from constants.

Reply:
Same thought I mentioned above.